File: 01/C++/01-1.cpp
#include "csv_helpers.hpp"

int main()
{
    std::string path = "/home/jonas/coding_practice/advent_of_code/01/advent_of_code.csv";
    std::vector<std::vector<int>> data = read_csv(path);

    std::vector<int> left_list{}, right_list{};
    std::tie(left_list, right_list) = extract_lists(data);

    int sum = 0;
    for (int i = 0; i < left_list.size(); i++)
    {
        int value = left_list[i];
        int value_counter = 0;
        for (int j = 0; j < left_list.size(); j++)
        {
            if (right_list[j] == value)
            {
                value_counter++;
            }
        }
        sum += value * value_counter;
    }
    std::cout << sum << "\n";
}

File: 01/python/01-1.py
import numpy as np

PATH = "/home/jonas/coding_practice/advent_of_code/01/"

both_lists = np.loadtxt(f"{PATH}advent_of_code.csv", delimiter=",", dtype=int)
first_list, second_list = np.split(both_lists, indices_or_sections=2, axis=1)

lenght_of_list = np.shape(first_list)[0]
first_list = np.reshape(first_list, shape=lenght_of_list)
second_list = np.reshape(second_list, shape=lenght_of_list)

number, counts = np.unique(second_list, return_counts=True)
number_and_counts = dict(zip(number, counts))

sum = 0
for i in range(lenght_of_list):
    key = first_list[i]
    if key in number_and_counts:
        sum += number_and_counts[key] * first_list[i]

print(sum)
